pkg/v1/config/clientconfig.go

```diff
@@ -273,14 +285,24 @@ func GetClientConfig() (cfg *configv1alpha1.ClientConfig, err error) {
 	addedDefaultDiscovery := populateDefaultStandaloneDiscovery(&c)
 	addedFeatureFlags := addDefaultFeatureFlagsIfMissing(&c, DefaultCliFeatureFlags)
 	addedEdition := addDefaultEditionIfMissing(&c)
+	addedCompatabilityFile := addCompatibilityFileIfMissing(&c)
 
-	if addedFeatureFlags || addedDefaultDiscovery || addedEdition {
+	if addedFeatureFlags || addedDefaultDiscovery || addedEdition || addedCompatabilityFile {
 		_ = StoreClientConfig(&c)
 	}
 
 	return &c, nil
 }
 
+// addDefaultEditionIfMissing returns true if the default edition was added to the configuration (because there was no edition)
+func addCompatibilityFileIfMissing(config *configv1alpha1.ClientConfig) bool {
+	if config.ClientOptions == nil || config.ClientOptions.CLI == nil || config.ClientOptions.CLI.CompatabilityFileLocation == "" {
+		addCompatabilityFile(config, resolveDefaultCompatabilityFile())
```

Comment:
If I am reading correctly, the linker-flag provided values will only be used here. But once the edition is set, there will be no way to go back to using those values unless one manually updates the compatibilityFileLocation config value. Is that the intended behavior?

Reply:
> But once the edition is set, there will be no way to go back to using those values unless one manually updates the compatibilityFileLocation config value. Is that the intended behavior?

I wasn't sure on this one. I do want to **bind** the edition change to a compatibility file change, but it seems like a valid use-case to change just the compatibility file.

Should I update this PR to offer a way to use `tanzu config` to change **just** the compatibility file?

Reply:
@joshrosso I think adding a `tanzu config` option is a good idea. Whether it should go in this PR or a different one I'll leave to Vui.

Reply:
Much of how relevant the use case is depends on how we handle the issue in the previous comment. Say the implementation is done such that TKG_CUSTOM_IMAGE_REPOSITORY will continue to be respected regardless of what is set in the config file, then there will be little need to update the compat file location via `tanzu config`.

Reply:
totally agreed. I'll follow-up on https://github.com/vmware-tanzu/tanzu-framework/pull/1605#discussion_r799852572 and we can go from there.